### PR TITLE
CFY-5652 Add conditional creation

### PR DESCRIPTION
**Pull request by szpotona**
Use `create_if_missing` property to control conditional creation of resources.

### CODE REVIEW

File: openstack_plugin_common/__init__.py [hunk 1]
```diff
@@ -212,9 +218,16 @@ def use_external_resource(ctx, sugared_client, openstack_type,
...
         return None
-
-    resource = _get_resource_by_name_or_id_from_ctx(
-        ctx, name_field_name, openstack_type, sugared_client)
+    resource = None
+    try:
+        resource = _get_resource_by_name_or_id_from_ctx(
+            ctx, name_field_name, openstack_type, sugared_client)
+    except NonRecoverableError:
+        if not resource and is_create_if_missing(ctx):
```
Comment (Jessiebr) [inline openstack_plugin_common/__init__.py]: unnecessary 'if not resource and...' 
can check only-
'if is_create_if_missing(ctx):'
...
else
...

File: openstack_plugin_common/__init__.py [hunk 2]
```diff
@@ -212,9 +218,16 @@ def use_external_resource(ctx, sugared_client, openstack_type,
...
+            ctx.instance.runtime_properties[CONDITIONALLY_CREATED] = True
+            return None
+        elif not resource:
```
Comment (Jessiebr) [inline openstack_plugin_common/__init__.py]: ctx.instance.runtime_properties[CONDITIONALLY_CREATED] = False

File: openstack_plugin_common/__init__.py (hunk 1 above)
Comment (szpotona) [inline openstack_plugin_common/__init__.py]: I agree

File: openstack_plugin_common/tests/openstack_client_tests.py [hunk 3]
```diff
@@ -258,3 +258,127 @@ def test_exceeded_quota(self):
...
+        self._test_validate_resource(True, True, False, client_mock)
+
+    def test_raise_error(self):
+        # If exists and shouldn't it is checked in resource
+        # validation so below scenario is not tested here
+        # self.assertRaises(NonRecoverableError,
+        #                   self._test_use_external_resource,
+        #                   is_external=False,
+        #                   create_if_missing=True,
+        #                   exists=True)
```
Comment (Jessiebr) [inline openstack_plugin_common/tests/openstack_client_tests.py]: erase comment

File: openstack_plugin_common/tests/openstack_client_tests.py [hunk 4]
```diff
@@ -258,3 +258,127 @@ def test_exceeded_quota(self):
...
+        self.assertIsNone(self._test_use_external_resource(True, True, False))
+
+    def test_raise_error(self):
+        # If exists and shouldn't it is checked in resource
+        # validation so below scenario is not tested here
+        # self.assertRaises(NonRecoverableError,
+        #                   self._test_use_external_resource,
+        #                   is_external=False,
+        #                   create_if_missing=True,
+        #                   exists=True)
```
Comment (Jessiebr) [inline openstack_plugin_common/tests/openstack_client_tests.py]: erase comment

Comment (szpotona) [inline openstack_plugin_common/tests/openstack_client_tests.py]: Comment created on purpose to cover (even silently) all True/False possibilities.

File: openstack_plugin_common/tests/openstack_client_tests.py (hunk 3 above)
Comment (szpotona) [inline openstack_plugin_common/tests/openstack_client_tests.py]: Comment created on purpose to cover (even silently) all True/False possibilities.

**Pull request merged**